core/src/main/java/hudson/lifecycle/UnixLifecycle.java

```diff
@@ -89,6 +87,10 @@ public void restart() throws IOException, InterruptedException {
 
     @Override
     public void verifyRestartable() throws RestartNotSupportedException {
+        if (!Functions.isGlibcSupported()) {
+            throw new RestartNotSupportedException("Restart is not supported on platforms without libc");
+        }
```

Comment:
The fix for JENKINS-72833: do not attempt to use `UnixLifecycle` if JNA can't make calls to `libc`.